File: abc/abc289/b.py
# https://atcoder.jp/contests/abc289/submissions/38830163

n, m = map(int, input().split())
a = set(map(int, input().split()))

st = []
for i in range(1, n + 1):
  st.append(i)

  if i not in a:
    while len(st) > 0:
      print(st.pop(), end = ' ')

File: abc/abc289/b.cpp
// https://atcoder.jp/contests/abc289/submissions/38791243

#include <bits/stdc++.h>
using namespace std;

int main() {
  int n, m;
  cin >> n >> m;

  vector<int> p = {0};
  for(int i = 1; i <= n; i++) p.push_back(i);

  vector<bool> b(n + 1, 0);
  int a;
  while(cin >> a) {
    b[a] = true;
  }

  stack<int> s;
  for(int i = 1; i <= n; i++) {
    if(!b[i]) {
      s.push(i);
      while(!s.empty()) {
        cout << s.top() << " ";
        s.pop();
      }
    } else {
      s.push(i);
    }
  }
}
